qiskit/ignis/mitigation/expval/ctmp_mitigator.py

```diff
@@ -0,0 +1,410 @@
+            for gen, rate in zip(self._generators, self._rates):
+                if qubits_set.issuperset(gen[2]):
+                    # Keep generator
+                    g_mat += rate * self._generator_to_coo_matrix(gen, qubits)
+
+            # Add generator matrix to cache
+            self._generator_mats[qubits] = sps.coo_matrix(g_mat)
+        return self._generator_mats[qubits]
+
+    def mitigation_matrix(self, qubits: List[int] = None) -> np.ndarray:
+        r"""Return the measurement mitigation matrix for the specified qubits.
+
+        The mitigation matrix :math:`A^{-1}` is defined as the inverse of the
+        :meth:`assignment_matrix` :math:`A`.
+
+        Args:
+            qubits: Optional, qubits being measured for operator expval.
+
+        Returns:
+            np.ndarray: the measurement error mitigation matrix :math:`A^{-1}`.
+        """
+        # NOTE: the matrix definition of G is somehow flipped in both row and
+        # columns compared to the canonical ordering for the A-matrix used
+        # in the Complete and Tensored methods
+        gmat = self.generator_matrix(qubits)
+        gmat = np.flip(gmat.todense())
+        return la.expm(-gmat)
+
+    def assignment_matrix(self, qubits: List[int] = None) -> np.ndarray:
+        r"""Return the measurement assignment matrix for specified qubits.
+
+        The assignment matrix is the stochastic matrix :math:`A` which assigns
+        a noisy measurement probability distribution to an ideal input
+        measurement distribution: :math:`P(i|j) = \langle i | A | j \rangle`.
+
+        Args:
+            qubits: Optional, qubits being measured for operator expval.
+
+        Returns:
+            np.ndarray: the assignment matrix A.
+        """
+        # NOTE: the matrix definition of G is somehow flipped in both row and
+        # columns compared to the canonical ordering for the A-matrix used
+        # in the Complete and Tensored methods
+        gmat = self.generator_matrix(qubits)
+        gmat = np.flip(gmat.todense())
+        return la.expm(gmat)
+
+    def noise_strength(self, qubits: Optional[int] = None) -> float:
+        """Return the noise strength :math:`gamma` on the specified qubits"""
+        if qubits is None:
+            qubits = tuple(range(self._num_qubits))
+        else:
+            qubits = tuple(sorted(qubits))
+
+        if qubits not in self._noise_strengths:
+            # Compute gamma and cache
+            g_mat = self.generator_matrix(qubits)
+            # Check ideal case
+            if g_mat.row.size == 0:
+                gamma = 0
+            else:
+                gamma = np.max(-g_mat.data[g_mat.row == g_mat.col])
+            if gamma < 0:
+                raise QiskitError(
+                    'gamma should be non-negative, found gamma={}'.format(gamma))
+            self._noise_strengths[qubits] = gamma
+        return self._noise_strengths[qubits]
+
+    def seed(self, value=None):
```

Comment:
Not sure that this is the correct location for this function

Reply:
This is how seeds are changed on persistent objects in the quantum info module, but i added an option to set fixed seed at init as well.